lib/ssz.ex

```diff
@@ -29,31 +29,43 @@ defmodule Ssz do
   ##### Utils
   defp error, do: :erlang.nif_error(:nif_not_loaded)
 
-  defp encode(%SszTypes.ExecutionPayloadHeader{} = map) do
-    Map.update!(map, :base_fee_per_gas, &encode_u256/1)
+  # Ssz types can have special decoding rules defined in their optional encode/1 function,
+  defp encode(%name{} = struct) do
+    case function_exported?(name, :encode, 1) do
+      true ->
+        name.encode(struct)
+
+      false ->
+        struct
+        |> Map.from_struct()
+        |> Enum.map(fn {k, v} -> {k, encode(v)} end)
+        |> then(&struct!(name, &1))
+    end
```

Comment:
Since the matched value is a bool, we can use an `if-else` here instead of a `case`.